src/clj/puppetlabs/puppetserver/ringutils.clj

```diff
@@ -85,6 +109,28 @@
         false))
     true))
 
+(schema/defn ^:always-validate request->name :- (schema/maybe schema/Str)
+  "Return the name of the client by parsing the subject of the client cert from
+  the request.  Return nil if the client cert is absent or cannot be parsed."
+  [request :- RingRequest]
+  (let [{:keys [ssl-client-cert]} request]
+    (if ssl-client-cert (get-cert-subject ssl-client-cert))))
+
+(schema/defn ^:always-validate authorized? :- schema/Bool
+  "(SERVER-768) replacement for client-allowed-access?"
+  [rules :- rules/Rules
+   request :- RingRequest]
+  (if-let [name (request->name request)]
```

Comment:
If you had a default "allow-all" rules.conf file and hit an endpoint with no SSL client certificate, should you be able to call into `rules/allowed?` with a `nil` value for `name` and have it return a truthy value for `authorized`?  Maybe `trapperkeeper-authorization` doesn't handle this today but seems like it should be capable of doing that.